### PR TITLE
AC-487-get-triggered-workflows-by-payload-api-endpoint

### PR DESCRIPTION
**Pull request by lillie-dae**
### Description

Ability to list Workflow Instance's for a given PayloadId
Also Fixes https://github.com/Project-MONAI/monai-deploy-workflow-manager/issues/411

### Status
**Ready**

### Types of changes
- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] All tests passed locally.
- [ ] [Documentation comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments) included/updated.

### Additional
**Example requests:**
http://localhost:5000/workflowinstances?payloadId=b732a1bc-aab6-45ce-a690-0ea6b9f86e17

http://localhost:5000/workflowinstances?status=created&payloadId=b732a1bc-aab6-45ce-a690-0ea6b9f86e17

http://localhost:5000/workflowinstances?status=created&payloadId=b732a1bc-aab6-45ce-a690-0ea6b9f86e17&pageSize=1000001

**Example Response:**

```json
{
    "pageNumber": 1,
    "pageSize": 2,
    "firstPage": "/workflowinstances?pageNumber=1&pageSize=2",
    "lastPage": "/workflowinstances?pageNumber=8&pageSize=2",
    "totalPages": 8,
    "totalRecords": 15,
    "nextPage": "/workflowinstances?pageNumber=2&pageSize=2",
    "previousPage": null,
    "data": [
        {
            "id": "0221291c-a994-4a7b-8ec6-42fbaa25185e",
            "ae_title": "Multi_Task_1",
            "workflow_id": "d662639b-30aa-4271-8fcf-b51e6745e72b",
            "payload_id": "b732a1bc-aab6-45ce-a690-0ea6b9f86e17",
            "start_time": "2022-09-06T09:36:03.995Z",
            "status": "Created",
            "bucket_id": "monai",
            "input_metadata": {
                "": ""
            },
            "tasks": [
                {
                    "execution_id": "da69dfb4-10bb-455d-bb6b-32d3eb5e993d",
                    "workflow_instance_id": "",
                    "task_type": "Multi_task",
                    "task_start_time": "0001-01-01T00:00:00Z",
                    "execution_stats": {},
                    "task_plugin_arguments": {},
                    "task_id": "b971c259-c3d5-4ddf-93b2-56c83fb7b474",
                    "previous_task_id": "",
                    "status": "Dispatched",
                    "reason": "None",
                    "input_artifacts": {},
                    "output_artifacts": {},
                    "output_directory": "payloadId/workflows/workflowInstanceId/executionId/",
                    "result": {},
                    "input_parameters": {},
                    "next_timeout": "0001-01-01T00:00:00Z",
                    "timeout_interval": 0
                }
            ]
        }
    ],
    "succeeded": true,
    "errors": null,
    "message": null
}


```

### CODE REVIEW

File: tests/IntegrationTests/WorkflowExecutor.IntegrationTests/Features/WorkflowInstancesApi.feature [hunk 1]
```diff
@@ -65,6 +65,32 @@ Scenario Outline: Get all workflow instances from API - Test pagination
...
+    Then I will get a 200 response
+    And Pagination is working correctly for the <pagination_count> workflow instances
+    And All results have correct <expected_status> and <expected_payloadId>
+    Examples:
+    | pagination_query                                           | amount | pagination_count | expected_status | expected_payloadId                   |
+    | ?pageSize=1                                                | 15     | 15               |                 |                                      |
+    | ?pageSize=1&status=created                                 | 15     | 15               | 0               |                                      |
+    | ?pageSize=1&payloadId=5450c3a9-2b19-45b0-8b17-fb10f89d1b2d | 15     | 15               |                 | 5450c3a9-2b19-45b0-8b17-fb10f89d1b2d |
+    #| ?pageNumber=10             | 15     | 15               |
+    #| ?pageNumber=1&pageSize=10  | 15     | 15               |
```
Comment (lillie-dae) [inline tests/IntegrationTests/WorkflowExecutor.IntegrationTests/Features/WorkflowInstancesApi.feature:82]: remove commented test

File: tests/IntegrationTests/WorkflowExecutor.IntegrationTests/Features/WorkflowInstancesApi.feature [hunk 2]
```diff
@@ -65,6 +65,32 @@ Scenario Outline: Get all workflow instances from API - Test pagination
     |                            | 0      | 0                |
     | ?pageNumber=1&pageSize=100 | 15     | 15               |
 
+@WorkflowInstancePagination2
```
Comment (lillie-dae) [inline tests/IntegrationTests/WorkflowExecutor.IntegrationTests/Features/WorkflowInstancesApi.feature:68]: remove 2

File: tests/IntegrationTests/WorkflowExecutor.IntegrationTests/StepDefinitions/WorkflowInstancesApiStepDefinitions.cs [hunk 3]
```diff
@@ -56,10 +56,31 @@ public void ThenICanSeeExpectedWorkflowInstanceIsReturned()
...
+            var request = ApiHelper.Request.RequestUri?.Query;
             var result = ApiHelper.Response.Content.ReadAsStringAsync().Result;
             var deserializedResult = JsonConvert.DeserializeObject<PagedResponse<List<WorkflowInstance>>>(result);
             Assertions.AssertPagination(count, request, deserializedResult);
         }
+
+        [Then(@"All results have correct (.*) and (.*)")]
+        public void AllResultsHaveExpectedStatusOrPayloadId(int? expected_status, string? expected_payloadId)
+        {
+            //var request = ApiHelper.Request.RequestUri?.Query;
```
Comment (lillie-dae) [inline tests/IntegrationTests/WorkflowExecutor.IntegrationTests/StepDefinitions/WorkflowInstancesApiStepDefinitions.cs:68]: remove comment

**Comment by jackschofield23** [comment]
Does this support the ability to disable pagination? I think Aide need the list of workflows all in one call. This could be done with another query param

**Review by neildsouth — APPROVED** [review]

**Review by jackschofield23 — CHANGES_REQUESTED** [review]
Does this support the ability to disable pagination? I think Aide need the list of workflows all in one call. This could be done with another query param

**Review by jackschofield23 — APPROVED** [review]

**Comment by RemakingEden** [comment]
Adding integration tests to this

**Pull request merged**